### PR TITLE
UI rework

### PR DESCRIPTION
**Pull request by mario**
This PR changes a significant portion of how the user interacts with the app. Please note that the PR is still WIP and I will apply the appropriate label once ready for a review.

- removes the bottom navigation bar
- switches settings to title tap
- changes the way you join a call via link (via the New conversations menu)
- changes all theming components to Material
- various cleanups & formatting changes
- other things, at least those mentioned here: https://github.com/nextcloud/talk-android/issues?q=is%3Aissue+is%3Aopen+label%3A%22pr+exists%22

### CODE REVIEW

**Comment by mario** [comment]
This is now ready for review @tobiasKaminsky and @AndyScherzinger. Thank you very much!

**Comment by mario** [comment]
FYI, Lint report is just wrong - it shows errors and warnings that are most definitely not there, probably from some previous runs when they were there.

**Comment by mario** [comment]
Ok, made it work! Ready for review @AndyScherzinger and @tobiasKaminsky.

**Comment by tobiasKaminsky** [comment]
This is a very very big PR, so it is quite hard to review it in depth.
So I will read the code and do a smoke test…

File: app/src/main/java/com/nextcloud/talk/controllers/ConversationsListController.java [hunk 1]
```diff
@@ -172,23 +184,41 @@ protected void onViewBound(@NonNull View view) {
...
+
+            GlideApp.with(getActivity())
+                    .asBitmap()
+                    .centerInside()
+                    .override(avatarSize, avatarSize)
+                    .apply(RequestOptions.bitmapTransform(new CircleCrop()))
+                    .load(glideUrl)
+                    .into(new SimpleTarget<Bitmap>() {
+                        @Override
+                        public void onResourceReady(@NonNull Bitmap resource, @Nullable Transition<? super Bitmap> transition) {
```
Comment (tobiasKaminsky) [inline app/src/main/java/com/nextcloud/talk/controllers/ConversationsListController.java]: What about onError() and a placeholder while loading and error image as fallback?
Or is this provided via ui / xml?

**Comment by tobiasKaminsky** [comment]
In general there are quite a few lines > 120 which makes it harder to review.

**Comment by tobiasKaminsky** [comment]
BUG:
- click "+"
- select "new public conversation"
- done
-> crash
- after restarting the app the public conversation is there and can be used
```
2018-12-10 18:03:18.263 13928-13928/com.nextcloud.talk2 E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.nextcloud.talk2, PID: 13928
    java.lang.NullPointerException: Attempt to invoke virtual method 'void com.kennyc.bottomsheet.BottomSheet.setOnShowListener(android.content.DialogInterface$OnShowListener)' on a null object reference
        at com.nextcloud.talk.controllers.ContactsController.prepareAndShowBottomSheetWithBundle(ContactsController.java:809)
        at com.nextcloud.talk.controllers.ContactsController.selectionDone(ContactsController.java:336)
        at com.nextcloud.talk.controllers.ContactsController.onOptionsItemSelected(ContactsController.java:364)
        at com.bluelinelabs.conductor.Controller.optionsItemSelected(Controller.java:1313)
        at com.bluelinelabs.conductor.Router.onOptionsItemSelected(Router.java:669)
        at com.bluelinelabs.conductor.internal.LifecycleHandler.onOptionsItemSelected(LifecycleHandler.java:272)
        at android.app.Fragment.performOptionsItemSelected(Fragment.java:2669)
```

File: app/src/main/java/com/nextcloud/talk/controllers/ConversationsListController.java (hunk 1 above)
Comment (mario) [reply app/src/main/java/com/nextcloud/talk/controllers/ConversationsListController.java]: None, but those didn't exist so far either. I might tackle this as part of move to Fresco (there's an issue for this).

**Comment by mario** [comment]
@tobiasKaminsky and @AndyScherzinger - please review. For some reason Drone can't get Flexbox, but works like a charm locally.

**Comment by tobiasKaminsky** [comment]
> For some reason Drone can't get Flexbox, but works like a charm locally.

I can also not build the app because Flexbox is missing.
I guess that it is working locally for you as you have a copy of it in ~/.gradle.
Delete it and try if it is working for you.

Maybe you need google() in your build.gradle in Chatkit?

**Comment by tobiasKaminsky** [comment]
Maybe you can/want also have a real Chatkit version instead of git hash?

Once I can compile it again, I'll test it.

**Comment by mario** [comment]
Google did some evil stuff by requiring all their stuff to be removed from
external repos.

I can do this tomorrow.

**Comment by tobiasKaminsky** [comment]
Shall I open up new issues for bugs, or post them here?

- create a 1:1 conversation
- enter conversation
- rotate screen
--> you now see the list with available conversations

- click on "+"
- see contact list
- rotate screen
- endless spinning wheel

**Comment by mario** [comment]
It's fine to have it here @tobiasKaminsky.

**Comment by tobiasKaminsky** [comment]
- join a conversation via link
- rotate, back to list (as above)
- but previously conversation is not shown, so again have to paste link

**Comment by mario** [comment]
@tobiasKaminsky both issues should now be fixed.

**Comment by tobiasKaminsky** [comment]
Should be three bugs, two in first comment, one in the next one.

Regarding lint: I disabled extra translation errors on translated languages, as those get removed by transifex anyways: https://github.com/nextcloud/android/blob/84303837a625fb800d94b864c25a657481f73246/lint.xml#L30-L33

Maybe you want to have a look at lint.xml and pick some useful…

**Comment by tobiasKaminsky** [comment]
All bugs that I found are solved :tada:

File: app/src/main/java/com/nextcloud/talk/controllers/ContactsController.java [hunk 2]
```diff
@@ -196,12 +200,7 @@ protected void onAttach(@NonNull View view) {
 
         if (isNewConversationView) {
             toggleNewCallHeaderVisibility(!isPublicCall);
-
-            checkAndHandleDoneMenuItem();
-
-            if (getActionBar() != null) {
-                getActionBar().setDisplayHomeAsUpEnabled(true);
-            }
+            //checkAndHandleDoneMenuItem();
```
Comment (tobiasKaminsky) [inline app/src/main/java/com/nextcloud/talk/controllers/ContactsController.java]: remove?

File: app/src/main/java/com/nextcloud/talk/controllers/ContactsController.java [hunk 3]
```diff
@@ -730,6 +675,21 @@ private void prepareViews() {
...
+
+    private void disengageProgressBar() {
+        if (!alreadyFetching) {
+            progressBar.setVisibility(View.GONE);
+            genericRvLayout.setVisibility(View.VISIBLE);
+
+            if (isNewConversationView) {
+                conversationPrivacyToogleLayout.setVisibility(View.VISIBLE);
+                joinConversationViaLinkLayout.setVisibility(View.VISIBLE);
+                //checkAndHandleDoneMenuItem();
```
Comment (tobiasKaminsky) [inline app/src/main/java/com/nextcloud/talk/controllers/ContactsController.java]: remove?

**Comment by tobiasKaminsky** [comment]
There seem to be a lot of unused resources that can be removed.

**Comment by tobiasKaminsky** [comment]
New Bug:

- long click on conversation
- see bottom sheet pop up
- rotate
- bottom sheet is gone
- long click on conversation -> nothing happens
- rotate back
- long click on conversation -> nothing happens

It should remember that bottom sheet is visible and it should be possible to toggle this always.

**Comment by mario** [comment]
Well honestly, that bug was always there, but I'll take a look and fix it @tobiasKaminsky. Thanks for the review!

**Comment by mario** [comment]
@tobiasKaminsky issue filled here #391 for looking-into later, so will merge this.

**Pull request merged**